Task_3/Task_3.2/MyDLL/DynamicArray.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MyDLL
+{
+    public class DynamicArray<T>: IEnumerable<T>, IEnumerable, ICloneable
+    {
+        private const int size = 8;
+
+        private T[] mainArray;
+
+        public int Capacity => mainArray.Length;
+
+        public int Length { get; private set; } = 0;
+
+        public DynamicArray()
+        {
+            mainArray = new T[size];
+        }
+
+        public DynamicArray(int capacity)
+        {
+            mainArray = new T[capacity];
+        }
+
+        public DynamicArray(IEnumerable<T> sourse) : this()
+        {
+            T[] tempArr = sourse.ToArray();
+            ReSize(tempArr.Length);
+
+            Array.Copy(tempArr, mainArray, tempArr.Length);
+        }
+
+        public void Add(T item)
+        {
+            if (Length+1 >= Capacity)
```

Comment:
```suggestion
            if (Length = Capacity)
```
`[1,2,3,4,5]` - capacity 5, length 5, next add needs a resized array